src/platform/Linux/DeviceInfoProviderImpl.h

```diff
@@ -27,6 +30,15 @@ class DeviceInfoProviderImpl : public DeviceInfoProvider
     DeviceInfoProviderImpl() = default;
```

Comment:
DeviceInfoProviderImpl is used in a way that is very hidden. I didn't even know it existed until today. This is not a platform item, it is a per-product item. We should move this away from PlatformManagerImpl and make Server have a reference to it, so that eventually we can init the UserLabel clutser to use the  DeviceInfoProvider rather than these deep hidden singletons

Reply:
DeviceInfoProviderImpl need to be implemented by each product and get registered to DeviceInfoProvider via SetDeviceInfoProvider, just like ConfigurationManagerImpl and DiagnosticDataProviderImpl

```
    SetConfigurationMgr(&ConfigurationManagerImpl::GetDefaultInstance());
    SetDiagnosticDataProvider(&DiagnosticDataProviderImpl::GetDefaultInstance());
    SetDeviceInfoProvider(&DeviceInfoProviderImpl::GetDefaultInstance()); 
```
Do you mean we don't need per-product implementation of DiagnosticDataProviderImpl, and move all implementations to DiagnosticDataProvider.cpp? Then we need to use standard interface to write/read all information currently provided by DeviceInfoProvider.

Reply:
I created https://github.com/project-chip/connectedhomeip/issues/17088 to track this since the scope of this refactor beyond this PR